src/game_actor.cpp

```diff
@@ -58,62 +58,66 @@ Game_Actor::Game_Actor(int actor_id) {
 	if (actor_id == 0) {
 		return;
 	}
+	dbActor = lcf::ReaderUtil::GetElement(lcf::Data::actors, GetId());
 
-	data.Setup(actor_id);
-	Setup();
+	data.two_weapon = dbActor->two_weapon;
+	data.lock_equipment = dbActor->lock_equipment;
+	data.auto_battle = dbActor->auto_battle;
+	data.super_guard = dbActor->super_guard;
 
-	const auto& skills = GetActor().skills;
-	for (auto& skill: skills) {
-		if (skill.level <= GetLevel()) {
-			LearnSkill(skill.skill_id, nullptr);
-		}
-	}
+	data.hp_mod = 0;
+	data.sp_mod = 0;
+	data.attack_mod = 0;
+	data.defense_mod = 0;
+	data.spirit_mod = 0;
+	data.agility_mod = 0;
 
-	RemoveInvalidData();
+	MakeExpList();
+	SetBattlePosition(GetOriginalPosition());
 
-	if (GetLevel() > 0) {
-		SetHp(GetMaxHp());
-		SetSp(GetMaxSp());
-		SetExp(exp_list[GetLevel() - 1]);
-	}
+	ChangeLevel(dbActor->initial_level, nullptr);
```

Comment:
Needs a test for COLORS: Lost Memories (there is a comment in this file somewhere) which uses Level 0.

Reply:
Retested and fixed a bug with this. Thanks for raising it.